src/StatusLine.php

```diff
@@ -0,0 +1,193 @@
+<?php
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+namespace Teapot;
+
+use Teapot\StatusCodeException\InvalidStatusCodeException;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * Interface representing a Value Object of the HTTP Status-Line, as
+ * specified in RFC 2616 and RFC 7231.
+ *
+ * PHP version 5.3
+ *
+ * @category StatusLine
+ *
+ * @package Teapot
+ *
+ * @author    Barney Hanlon <barney@shrikeh.net>
+ * @copyright 2013-2016 B Hanlon. All rights reserved.
+ * @license   MIT http://opensource.org/licenses/MIT
+ * @link      https://shrikeh.github.com/teapot
+ */
+class StatusLine
```

Comment:
@shrikeh 

On the other hand, at the moment it's a bit weird because both

* `StatusCode` and
* `StatusCode\StatusCodeInterface`

are interfaces, what do you think?